File: app/src/main/java/br/com/algorit/mangaalert/roomdatabase/model/Novel.java
package br.com.algorit.mangaalert.model;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "NOVEL")
public class Novel {

    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "NOME")
    private String nome;

    @ColumnInfo(name = "CAPITULO")
    private Double capitulo;

    @ColumnInfo(name = "CHECKED")
    private boolean checked;

    @Ignore
    public Novel(@NonNull String nome) {
        this.setNome(nome);
    }

    public Novel(@NonNull String nome, Double capitulo, boolean checked) {
        this.nome = nome;
        this.capitulo = capitulo;
        this.checked = checked;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Double getCapitulo() {
        return capitulo;
    }

    public void setCapitulo(Double capitulo) {
        this.capitulo = capitulo;
    }

    public boolean isChecked() {
        return checked;
    }

    public void setChecked(boolean checked) {
        this.checked = checked;
    }
}

File: app/src/main/java/br/com/algorit/mangaalert/roomdatabase/model/Novel.kt
package br.com.algorit.mangaalert.roomdatabase.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity(tableName = "NOVEL")
class Novel {
    @PrimaryKey
    @ColumnInfo(name = "NOME")
    var nome: String

    @ColumnInfo(name = "CAPITULO")
    var capitulo: Double? = null

    @ColumnInfo(name = "CHECKED")
    var isChecked = false

    @Ignore
    constructor(nome: String) {
        this.nome = nome
    }

    constructor(nome: String, capitulo: Double?, checked: Boolean) {
        this.nome = nome
        this.capitulo = capitulo
        isChecked = checked
    }

    operator fun compareTo(novel: Novel): Int {
        return if (isChecked && !novel.isChecked) {
            -1
        } else if (!isChecked && novel.isChecked) {
            1
        } else {
            0
        }
    }
}